src/cache/cache_las.c

```diff
@@ -1068,8 +1082,7 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
          * all we wanted or enough and there is a reader waiting and we're on a key boundary.
          */
         ++visit_cnt;
-        if (!removing_key_block &&
-          (cnt == 0 || (visit_cnt > WT_LAS_SWEEP_ENTRIES && cache->las_reader)))
+        if (cnt == 0 || (visit_cnt > WT_LAS_SWEEP_ENTRIES && cache->las_reader))
```

Comment:
>   __wt_buf_free(session, sweep_key);
    ret = WT_NOTFOUND;
    break;

Using WT_ERR there generates an error, but that scenario is a success scenario, later the return value is modified to 0. So I changed to `goto err;`

Reply:
Good catch. I think there still may be something here, because the `err` label is going to free `sweep_key`. Should we jump to `srch_notfound` instead?